### ISSUE TITLE
Missing Data::Dumper from list of dependencies

### ISSUE THREAD

**Issue by ppisar** [Test-More/Test2-Suite#154]
Test2-Suite-0.000102 added "require Data::Dumper" into lib/Test2/AsyncSubtest.pm:111. If Data::Dumper is not available, t/modules/AsyncSubtest.t test fails because of that:

~~~
# Failed test 'When an event is seen to early we get to see it in our exception'
# at t/modules/AsyncSubtest.t line 201.
# +--------------------------------+----+--------------------------------+
# | GOT                            | OP | CHECK                          |
# +--------------------------------+----+--------------------------------+
# | Can't locate XXX/Data/Dumper.p | =~ | (?^:We want to see this messag |
# | m in @INC (you may need to ins |    | e or people die!)              |
# | tall the XXX::Data::Dumper mod |    |                                |
# | ule) (@INC contains: /home/tes |    |                                |
# | t/fedora/perl-Test2-Suite/Test |    |                                |
# | 2-Suite-0.000102/blib/lib /hom |    |                                |
# | e/test/fedora/perl-Test2-Suite |    |                                |
# | /Test2-Suite-0.000102/blib/arc |    |                                |
# | h /usr/local/lib64/perl5 /usr/ |    |                                |
# | local/share/perl5 /usr/lib64/p |    |                                |
# | erl5/vendor_perl /usr/share/pe |    |                                |
# | rl5/vendor_perl /usr/lib64/per |    |                                |
# | l5 /usr/share/perl5 .) at /hom |    |                                |
# | e/test/fedora/perl-Test2-Suite |    |                                |
# | /Test2-Suite-0.000102/blib/lib |    |                                |
# | /Test2/AsyncSubtest.pm line 11 |    |                                |
# | 1.\n                           |    |                                |
# +--------------------------------+----+--------------------------------+

# Failed test 'We see the main exception'
# at t/modules/AsyncSubtest.t line 207.
# +--------------------------------+----+--------------------------------+
# | GOT                            | OP | CHECK                          |
# +--------------------------------+----+--------------------------------+
# | Can't locate XXX/Data/Dumper.p | =~ | (?^:You must attach to an Asyn |
# | m in @INC (you may need to ins |    | cSubtest before you can send e |
# | tall the XXX::Data::Dumper mod |    | vents to it from another proce |
# | ule) (@INC contains: /home/tes |    | ss or thread)                  |
# | t/fedora/perl-Test2-Suite/Test |    |                                |
# | 2-Suite-0.000102/blib/lib /hom |    |                                |
# | e/test/fedora/perl-Test2-Suite |    |                                |
# | /Test2-Suite-0.000102/blib/arc |    |                                |
# | h /usr/local/lib64/perl5 /usr/ |    |                                |
# | local/share/perl5 /usr/lib64/p |    |                                |
# | erl5/vendor_perl /usr/share/pe |    |                                |
# | rl5/vendor_perl /usr/lib64/per |    |                                |
# | l5 /usr/share/perl5 .) at /hom |    |                                |
# | e/test/fedora/perl-Test2-Suite |    |                                |
# | /Test2-Suite-0.000102/blib/lib |    |                                |
# | /Test2/AsyncSubtest.pm line 11 |    |                                |
# | 1.\n                           |    |                                |
# +--------------------------------+----+--------------------------------+
# Seeded srand with seed '20180305' from local date.
t/modules/AsyncSubtest.t ........................ Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/14 subtests
~~~

Please add Data::Dumper to test and run dependency list.

**Comment by exodist** [comment]
This is strange in a lot of ways.

1) it is trying to find XXX/Data/Dumper.pm in your error, not Data::Dumper.
2) Data::Dumper is a core module, so technically it does not need to be listed.

That said, I will still add Data::Dumper to the deps. I am just not certain it is going to help with your problem.

**Comment by ppisar** [comment]
On Mon, Mar 05, 2018 at 05:25:38PM +0000, Chad Granum wrote:
> 1) it is trying to find XXX/Data/Dumper.pm in your error, not Data::Dumper.

Because I rewrote the Data::Dumper into XXX::Data::Dumper in the Test-Suite
code to verify it is really used by that code. It's the easiest and safer way.
And the paste the error here.

> 2) Data::Dumper is a core module, so technically it does not need to be listed.
> 
You never know when a module will be removed from Perl. And you already
list other core modules in the dependencies. So I thought it was an omission
and I reported it.

> That said, I will still add Data::Dumper to the deps. I am just not certain
> it is going to help with your problem.
> 
Yes, it helps. Thank you.

-- Petr
